src-cljs/views/head-form.cljs

```diff
@@ -0,0 +1,62 @@
+(ns doctopus.views.head-form
+  (:require [cljs.core.async :refer [<!]]
+            [cljs-http.client :as http]
+            [doctopus.util :refer [get-value http-ok? redirect-to in?]]
+            [doctopus.views.common :refer [button]]
+            [reagent.core :as reagent :refer [atom]])
+            (:require-macros [cljs.core.async.macros :refer [go]]))
+
+(def form-data (atom {}))
+(def csrf-token (atom ""))
+(def existing-heads (atom []))
+(def errors (atom []))
```

Comment:
the idea here was to have a "global" `errors` vector that would be populated (or emptied!) as a reaction to the `form-data` atom changing. This was probably misguided, however, and should be implemented via a single atom to rule them all.